### PR TITLE
Update documentation for `pillarbox-analytics`

### PR DESCRIPTION
**Pull request by MGaetan89**
# Pull request

## Description

Update the documentation for `pillarbox-analytics`.

## Changes made

- Update `PillarboxAndroidLibraryPublishingPlugin` to support loading module documentation from either `Module.md` or `docs/README.md`.
- Update `docs/README.md` to work with Dokka (mainly removing images) and improving wording and samples.
- Remove `Module.md`.
- Update the KDoc for every public class/method/interface.
- Make `TCEventExtensions` internal. I don't think this was meant to be public. Let me know if I should change it back.

## Checklist

- [x] APIs have been properly documented (if relevant).
- [x] The documentation has been updated (if relevant).
- [ ] New unit tests have been written (if relevant).
- [ ] The demo has been updated (if relevant).

### CODE REVIEW

File: pillarbox-analytics/docs/README.md [hunk 1]
```diff
@@ -1,95 +1,122 @@
...
-### Configuration and create
+### Configure analytics
 
-Before using `SRGAnalytics` make sure to call `SRGAnalytics.init` first, otherwise it can lead to undefined behavior.
-It is strongly recommended to call the initializer inside your `Application.onCreate` method.
+Before using any functionality, `SRGAnalytics` must be initialized in your [Application][android.app.Application]'s
+[onCreate()][android.app.Application.onCreate] method using either the
+[initSRGAnalytics()][ch.srgssr.pillarbox.analytics.SRGAnalytics.initSRGAnalytics] or the
+[SRGAnalytics.init()][ch.srgssr.pillarbox.analytics.SRGAnalytics.init] method and providing an
+[AnalyticsConfig][ch.srgssr.pillarbox.analytics.AnalyticsConfig] instance.
```
Comment (StaehliJ) [inline pillarbox-analytics/docs/README.md:24]: It doesn't display well on the browser. Links should be added manually here but dokka get theme automatically.

File: pillarbox-analytics/docs/README.md [hunk 2]
```diff
@@ -1,95 +1,122 @@
...
-class DemoApplication : Application() {
-
+class MyApplication : Application() {
     override fun onCreate() {
         super.onCreate()
-        
+
         val config = AnalyticsConfig(
             vendor = AnalyticsConfig.Vendor.SRG,
-            nonLocalizedApplicationName = "PillarboxDemo",
```
Comment (StaehliJ) [inline pillarbox-analytics/docs/README.md:33]: We should keep the `nonLocalizedApplicationName` as it is useful for apps that localized the application name.

`nonLocalizedApplicationName = `You application name not localized here`

File: pillarbox-analytics/docs/README.md [hunk 3]
```diff
@@ -1,95 +1,122 @@
...
+
+Top-level entry point for managing analytics in Pillarbox for SRG SSR applications.
+
+# Package ch.srgssr.pillarbox.analytics.commandersact
+
+Commanders Act specific classes.
+
+# Package ch.srgssr.pillarbox.analytics.comscore
+
+ComScore specific classes.
```
Comment (StaehliJ) [inline pillarbox-analytics/docs/README.md:122]: I assume it is dokka specific ?

**Review by StaehliJ — CHANGES_REQUESTED** [review]

File: pillarbox-analytics/docs/README.md (hunk 1 above)
Comment (MGaetan89) [reply pillarbox-analytics/docs/README.md:24]: It should be good now: https://github.com/SRGSSR/pillarbox-android/blob/revisit_analytics_documentation/pillarbox-analytics/docs/README.md

(depending on the merge of #775)

File: pillarbox-analytics/docs/README.md (hunk 3 above)
Comment (MGaetan89) [reply pillarbox-analytics/docs/README.md:124]: Indeed. The format is documented here: https://kotlinlang.org/docs/dokka-module-and-package-docs.html

A simple example:
```markdown
# Module my-module

The documentation for the `my-module` module.

# Package com.example.my.package

The documentation for the `com.example.my.package` package.
```

**Review by StaehliJ — APPROVED** [review]

**Pull request merged**